### ISSUE TITLE
Sprint Oct 19th

### ISSUE THREAD

**Issue by RichardLitt** [ipfs/pm#40]
# Sprint Oct 19th

## Sprint Goals

- _TBD_

## Sprint Discussions

See https://github.com/ipfs/pm#sprint-discussion-schedule

Please take notes in a separate pad, if you can, and link it here.

- node-ipfs notes:  
- go-ipfs notes:  
- ipfs-app notes:  
- infrastructure: https://pad.okfn.org/p/ipfs-infrastructure

## Sprint Deliverables

- Add your checklists below - that way we only have people listed who are really in the sprint. You can add them to this list, or as separate comments in this issue. 
- Add links to issues down here. Only add things you can finish this sprint.

### @lgierth

- [ ] move pollux to protocol labs account https://github.com/ipfs/infrastructure/issues/73
- [ ] gateway-dmca-denylist
- [ ] another storage host with hetzner https://github.com/ipfs/infrastructure/issues/89
- [ ] invite to fc00 development (aka go-cjdns)
- [ ] spec skeletons for fc00 parts
- [x] have hostnames in grafana, instead of cjdns addresses ipfs/infrastructure#102
- [x] deploy golang1.5.1 in order to fix pinbot ipfs/infrastructure#103
- [x] fix api timeouts manifesting with pinbot ipfs/infrastructure#104

### @kyledrake

- [x] 0.4.0 blog post
- [x] Electron app announcement post (https://github.com/ipfs/electron-app/issues/46)
- [ ] Refresh with jbenet on blog
- [ ] Discuss with jbenet plans for DMCA
- [ ] DMCA info/takedown contact page for ipfs.io
- [ ] Dynamically wire libp2p markdown specs into docs for libp2p site
- [ ] Work on new IPFS talk

### @jbenet

- [ ] SF meetup
- [ ] [ee380 Stanford talk](http://ee380.stanford.edu/Abstracts/151021.html)
- [ ] org things.

### @chriscool

- [x] add flag coverage to Sharness coverage script
- [ ] polish Sharness coverage script
- [x] finish split between long and short options

### @whyrusleeping 

- [ ] fix windows builds https://github.com/ipfs/go-ipfs/pull/1875
- [ ] fix streaming output https://github.com/ipfs/go-ipfs/pull/1882
- [ ] replace go1.5 requirement https://github.com/ipfs/go-ipfs/pull/1882
- [ ] merge utp https://github.com/ipfs/go-ipfs/pull/1789
- [ ] fix running ipfs id on self https://github.com/ipfs/go-ipfs/pull/1856
- [ ] rebase 0.4.0 (after utp merges!)
- [ ] write ipfs-update https://github.com/ipfs/ipfs-update
- [ ] write cool docker tests
  - [x] lib to control networking https://github.com/whyrusleeping/go-ctrlnet
  - [ ] write some cool benchmarks using docker and ctrlnet https://github.com/whyrusleeping/ipfs-network-tests
- [ ] fix bug in go-log https://github.com/ipfs/go-ipfs/issues/1874
- [x] a few new calls in go-ipfs-api
- [ ] cache results of ipns name resolution for a short time https://github.com/ipfs/go-ipfs/pull/1887
- [ ] ipfs swarm connect should error when it fails https://github.com/ipfs/go-ipfs/pull/1900

**Comment by RichardLitt** [comment]
Goal this week:
- [ ] ipfs HTTP documentation

**Comment by Dignifiedquire** [comment]
### Sprint Update by @dignifiedquire

#### Station

- [x] Create first simple packages
- [x] Research on better packaging
- [~] Fix packaging issues and create native installers (dmg, win installer, deb, rpm)

### node-ipfs-api

- [~] Move to superagent to fix random `ECONNREST` issues
- [~] Work on proper test setup (browser & node)

**Comment by diasdavid** [comment]
### SPRINT UPDATE

- [~] node-ipfs-api solid tests, a lot of work going on, thanks to @dignifiedquire and @victorbjelkholm for chipping in on this one https://github.com/ipfs/node-ipfs-api/pull/81
- [x] San Francisco and Palo Alto IPFS events stravaganza 
- [x] Lots of flying
- [~] webrtc-explorer as one of the transports of libp2p (ongoing)
- [ ] Speed up Jeromy on how libp2p is modularised
- [x] Set up analytics things in Sum All
- [x] List articles, talks and other artefacts of IPFS presence at https://github.com/ipfs/website/pull/71
- [x] get ipfsd-ctl to fire daemons without connecting to the central network
- [x] complement ipscend README with dnslink
- [x] add coverflow to ipscend preview (still needs a proper designer hands though)
- [x] logistics for San Francisco IPFS meetup (food, bannerman, etc)	
- [ ] decide with @jbenet the articles roadmap for libp2p

Was blocked on getting npm's registry on IPFS due to lack of machine with enough HDD and stable internet link (from moving to place to place constantly).

**Comment by lgierth** [comment]
- first week after time off, worked through a backlog of email and chat
- storage
  - [~] move pollux to protocol labs account (waiting for feedback) https://github.com/ipfs/infrastructure/issues/73
  - [~] another storage host with hetzner https://github.com/ipfs/infrastructure/issues/89
- [~] gateway-dmca-denylist
- fc00/cjdns
  - [~] invite to fc00 development (aka go-cjdns)
  - [ ] spec skeletons for fc00 parts
- pinbot
  - [x] deploy golang1.5.1 in order to fix pinbot ipfs/infrastructure#103
  - [x] fix api timeouts manifesting with pinbot ipfs/infrastructure#104
- metrics
  - [x] more graphs
  - [x] have hostnames in grafana, instead of cjdns addresses ipfs/infrastructure#102

storage needs feedback (https://github.com/protocol/logistics/issues/21 && https://github.com/ipfs/infrastructure/issues/89), dmca PR is about to fly in today, same for the fc00 invitation

**Comment by VictorBjelkholm** [comment]
### Update

IPFS in JS

- [~] working on node-ipfs-api to pass the tests in browser + node env
- [ ] find and help out with things to contribute to node-ipfs (will need help from daviddias with this)

OpenIPFS - https://github.com/victorbjelkholm/openipfs

- [~] having openipfs ready for public consumption, mainly to scratch my own need for ipfsbin

IPFSBin - https://github.com/victorbjelkholm/ipfsbin

- [~] new ui for ipfsbin to be able to select language amongst other things
- [ ] start using openipfs for ipfsbin, to make content more available

Community

- [ ] preparing talk for local meetups here in Barcelona 
- [ ] collecting things for awesome-ipfs showcase

**Comment by kyledrake** [comment]
Update

- [x] 0.4.0 blog post
- [x] Electron app announcement post (https://github.com/ipfs/electron-app/issues/46)
- [ ] Refresh with jbenet on blog / DMCA
- [~] DMCA info/takedown contact page for ipfs.io
- [~] Dynamically wire libp2p markdown specs into docs for libp2p site
- [ ] Work on new IPFS talk

**Comment by whyrusleeping** [comment]
Got a lot of PRs opened up this week, lots of bugfixes
- [ ] fix windows builds https://github.com/ipfs/go-ipfs/pull/1875
- [ ] fix streaming output https://github.com/ipfs/go-ipfs/pull/1882
- [ ] replace go1.5 requirement https://github.com/ipfs/go-ipfs/pull/1882
- [ ] merge utp https://github.com/ipfs/go-ipfs/pull/1789
- [ ] fix running ipfs id on self https://github.com/ipfs/go-ipfs/pull/1856
- [ ] rebase 0.4.0 (after utp merges!)
- [ ] write ipfs-update https://github.com/ipfs/ipfs-update
- [ ] write cool docker tests
  - [x] lib to control networking https://github.com/whyrusleeping/go-ctrlnet
  - [ ] write some cool benchmarks using docker and ctrlnet https://github.com/whyrusleeping/ipfs-network-tests
- [ ] fix bug in go-log https://github.com/ipfs/go-ipfs/issues/1874
- [x] a few new calls in go-ipfs-api
- [ ] cache results of ipns name resolution for a short time https://github.com/ipfs/go-ipfs/pull/1887
- [ ] ipfs swarm connect should error when it fails https://github.com/ipfs/go-ipfs/pull/1900
ipfs update getting polished and sanity checks.

**Comment by RichardLitt** [comment]
[ ] IPFS API documentation: Trying to figure out how to get swagger integrated into go-ipfs. Not entirely clear on this, any help would be great.

**Comment by gamemanj** [comment]
jbenet said during the IRC discussion I should link to Mewler here, so: /ipns/QmarsUDwQ5ak34KU6xJbr2iMuatB7e69WfuSjEBG3v7zvu/
And by hash: /ipfs/QmYgbBTcBb9KqRSRZjbu6AonY9Y86GEStm8hvofqnqbkBV/
I'll put it up on GitHub soon, probably as gamemanj/Mewler, after removing the hopefully confusing april fool's joke.

**Comment by RichardLitt** [comment]
New issue: https://github.com/ipfs/pm/issues/41

**Comment by jbenet** [comment]
I was out most of this week with some important visits/meetings in SF.

- [x] IPFS meetup on tue -- thanks Sourcegraph! videos coming
- [x] Stanford visit
  - [x] talks with profs
  - [x] talk at ee380 - https://www.youtube.com/watch?v=HUVmypx9HGI
- [x] tons of important org things
